src/components/side-panel/layer-panel/layer-column-config.js

```diff
@@ -22,118 +22,128 @@ import React, {Component} from 'react';
 import PropTypes from 'prop-types';
 import styled from 'styled-components';
 import {FormattedMessage} from 'localization';
-import FieldSelector from 'components/common/field-selector';
 import {createSelector} from 'reselect';
 
 import {PanelLabel, SidePanelSection} from 'components/common/styled-components';
+import FieldSelectorFactory from '../../common/field-selector';
 
 const TopRow = styled.div`
   display: flex;
   justify-content: space-between;
 `;
 
-export default class LayerColumnConfig extends Component {
-  static propTypes = {
-    columns: PropTypes.object.isRequired,
-    fields: PropTypes.arrayOf(PropTypes.any).isRequired,
-    assignColumnPairs: PropTypes.func.isRequired,
-    assignColumn: PropTypes.func.isRequired,
-    updateLayerConfig: PropTypes.func.isRequired,
-    columnPairs: PropTypes.object,
-    fieldPairs: PropTypes.arrayOf(PropTypes.any),
-    columnLabels: PropTypes.object
-  };
+const LayerColumnConfigFactory = ColumnSelector => {
```

Comment:
same here, make it a function instead of const ...
put the deps definition on top of the function declaration for easy read.

Reply:
Done